tests/useMultiRootEditor.test.tsx

```diff
@@ -931,16 +931,20 @@ describe( 'useMultiRootEditor', () => {
 					editor: SlowEditor
 				} ) );
 
-				await timeout( 100 );
+				await timeout( 500 );
 
+				// Depending on the execution order on the event loop this `setData` might be delayed by the React engine.
+				// It happens only if event loop is busy and React has to wait for the next tick a little bit longer than usual.
+				// It does not play well with `waitFor` below so we added few retries to make it more stable.
+				// It should not be a problem in real life, as it is a rare case and might be solved in future React versions.
 				result.current.setData( {
 					intro: 'Hello World!',
 					content: ''
 				} );
 
-				await timeout( 200 );
+				await timeout( 500 );
```

Comment:
It's not super elegant, but I understand the problems with React.

Reply:
Yep :[